pkg/apis/hypercloud/v1alpha1/virtualmachinevolume_types.go

```diff
@@ -33,18 +33,43 @@ const (
 	VirtualMachineVolumeStateError VirtualMachineVolumeState = "Error"
 )
 
-// VirtualMachineVolumeStatus defines the observed state of VirtualMachineVolume
+// VirtualMachineVolumeConditionType is a string representation of known VirtualMachineVolume conditions
+type VirtualMachineVolumeConditionType string
+
+const (
+	// ReadyToUse indicates VirtualMachineVolume is ready to use
+	ReadyToUse VirtualMachineVolumeConditionType = "ReadyToUse"
+)
+
+// VirtualMachineVolumeCondition contains details of the current condition of a VirtualMachineVolume.
+type VirtualMachineVolumeCondition struct {
+	// Type of VirtualMachineVolume condition
+	Type   VirtualMachineVolumeConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown
+	Status corev1.ConditionStatus            `json:"status"`
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+	// One-word CamelCase reason for the condition's last transition
+	// +optional
+	Reason string `json:"reason,omitempty"`
+	// Human-readable message indicating details about last transition.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// VirtualMachineVolumeStatus defines the observed status of VirtualMachineVolume
 type VirtualMachineVolumeStatus struct {
-	// State is the current state of VirtualMachineVolume
 	State VirtualMachineVolumeState `json:"state"`
+	Conditions []VirtualMachineVolumeCondition `json:"conditions,omitempty"`
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // VirtualMachineVolume is the Schema for the virtualmachinevolumes API
 // +kubebuilder:subresource:status
 // +kubebuilder:resource:path=virtualmachinevolumes,scope=Namespaced,shortName=vmv
-// +kubebuilder:printcolumn:name="State",type="string",JSONPath=".status.state",description="Current state of VirtualMachineVolume"
+// +kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.conditions[?(@.type==\"ReadyToUse\")].status",description="Current status of VirtualMachineVolume"
```

Comment:
state gives an overview of the conditions. The State is better for showing to the user

Reply:
done